File: src/main/java/ru/hollowhorizon/hc/common/registry/ModTileEntities.java
package ru.hollowhorizon.hc.common.registry;

import net.minecraft.world.level.block.entity.BlockEntityType;
import ru.hollowhorizon.hc.api.registy.HollowRegister;
import ru.hollowhorizon.hc.common.objects.tiles.SaveObeliskTile;

public class ModTileEntities {
    @HollowRegister()
    public static final BlockEntityType<SaveObeliskTile> SAVE_OBELISK_TILE = BlockEntityType.Builder.of(
            SaveObeliskTile::new, ModBlocks.SAVE_OBELISK_BLOCK
    ).build(null);
}

File: src/main/java/ru/hollowhorizon/hc/common/registry/ModTileEntities.kt
package ru.hollowhorizon.hc.common.registry

import net.minecraft.world.level.block.entity.BlockEntityType
import ru.hollowhorizon.hc.common.objects.tiles.SaveObeliskTile

object ModTileEntities : HollowRegistry() {
    val SAVE_OBELISK_TILE by register {
        BlockEntityType.Builder.of(
            ::SaveObeliskTile,
            ModBlocks.SAVE_OBELISK_BLOCK.get()
        ).build(promise())
    }
}